src/simpleNbox.cpp

```diff
@@ -53,28 +53,29 @@ void SimpleNbox::init( Core* coreptr ) {
     tempferts[ SNBOX_DEFAULT_BIOME ] = 1.0;
 
     Tgav_record.allowInterp( true );
-    
+
     // Register the data we can provide
     core->registerCapability( D_ATMOSPHERIC_CO2, getComponentName() );
     core->registerCapability( D_ATMOSPHERIC_C, getComponentName() );
     core->registerCapability( D_PREINDUSTRIAL_CO2, getComponentName() );
     core->registerCapability( D_RF_T_ALBEDO, getComponentName() );
     core->registerCapability( D_LAND_CFLUX, getComponentName() );
-    
+
     // Register our dependencies
     core->registerDependency( D_OCEAN_CFLUX, getComponentName() );
 
     // Register the inputs we can receive from outside
     core->registerInput(D_FFI_EMISSIONS, getComponentName());
     core->registerInput(D_LUC_EMISSIONS, getComponentName());
-    core->registerInput(D_PREINDUSTRIAL_CO2, getComponentName()); 
+    core->registerInput(D_PREINDUSTRIAL_CO2, getComponentName());
     core->registerInput(D_BETA, getComponentName());
     core->registerInput(D_Q10_RH, getComponentName());
     core->registerInput(D_F_NPPV, getComponentName());
     core->registerInput(D_F_NPPD, getComponentName());
     core->registerInput(D_F_LITTERD, getComponentName());
     core->registerInput(D_F_LUCV, getComponentName());
     core->registerInput(D_F_LUCD, getComponentName());
+    core->registerInput(D_CA_CONSTRAIN, getComponentName());
```

Comment:
@rplzzz Okay but how would Hector distinguish if the CO2 had been constrained or normal atmospheric CO2 output? Would the core keep track of `CA_CONSTRAIN` if it was set during setvars?

Reply:
If no constraints have been set, then the constraint array will exist, but it will be empty, so that provides a way to tell.

I think a more challenging issue is that the Hector API requires us to specify a date for the constraint that we're requesting.  There's no way to ask what dates are available.   I have some ideas on how to solve this, which would also solve another annoying API feature, which is that we can't request variables for times later than the current time, even though for things like emissions those variables often exist for times that haven't been run yet.  We could go over those ideas the next time we have a chance to talk.